apps/explorer/src/api/assets/assetsApi.ts

```diff
@@ -167,7 +167,15 @@ export const assetsQueries = {
         queryFn: async (): Promise<NFTTokenUriMetaData & { assetId: string }> => {
           const nftData = (await axios.get(dataUri)).data as NFTTokenUriMetaData
 
-          return matchesNFTTokenUriMetaDataSchema(nftData) ? { ...nftData, assetId } : Promise.reject()
+          return matchesNFTTokenUriMetaDataSchema(nftData)
+            ? { ...nftData, assetId }
+            : !!nftData?.name || !!nftData?.image
+              ? {
+                  assetId,
+                  name: nftData?.name ? nftData?.name.toString() : 'Unsupported NFT',
+                  image: nftData?.image ? nftData?.image.toString() : ''
```

Comment:
The second optional chaining operator in each line is not needed

Reply:
I'm not sure I understand. Trying to run `toString()` on `undefined` will throw an error

Reply:
```suggestion
                  name: nftData?.name ? nftData.name.toString() : 'Unsupported NFT',
                  image: nftData?.image ? nftData.image.toString() : ''
```

this should work

Reply:
Ah! I get it now :)

Reply:
Done